### ISSUE TITLE
Bug: Optionals parameters on mock drivers raise an error

### ISSUE THREAD

**Issue by cacique-coder** [PlaceOS/drivers#193]
**Describe the bug**

When writing a driver spec, given a mock method with a optional parameter, if the same parameter is omitted in the main code, raise the next following error.

Missing hash key: "title" (KeyError) (PlaceOS::Driver::RemoteException)
  from /usr/share/crystal/src/hash.cr:1027:11 in '[]'
  from /usr/share/crystal/src/json/any.cr:102:7 in '[]'
  from repositories/wcp-drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:148:62 in '__build_request__'
  from repositories/wcp-drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:95:17 in '__exec_request__'
  from repositories/wcp-drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:12:5 in '???'
  from repositories/wcp-drivers/drivers/events/booking_notification.cr:56:22 in 'check_booking'

**To Reproduce**

With any spec, writing a `DriverSpecs::MockDriver`
```rb
#spec
require "placeos-driver/interface/mailer"

class StaffAPI < DriverSpecs::MockDriver
  def get_event(id : String, system_id : String? = nil, calendar_id : String? = nil)
  end
end

#driver 
class Desk::AnyDriver< PlaceOS::Driver
    accessor staff_api : StaffAPI_1

    def any_call
      staff_api.get_event("id", "system_id").get
    end
```

**Screenshots or a paste of terminal output**

```
Launching spec runner
Launching driver: /app/bin/drivers/any_driver_HEAD
... requesting default settings
... got default settings: {"timezone" => "Australia/Brisbane", "date_time_format" => "%c", "time_format" => "%l:%M%p", "date_format" => "%A, %-d %B", "booking_type" => "event"}
... starting driver IO services
... starting module
... waiting for module
ignoring command start in driver-runner server
... module connected
... enabling debugging output
... starting spec
-> spec calling: any_call {}
level=[W] time=2021-06-30T06:25:47Z program=any_driver_HEAD message="Deprecated metadata in use. A driver update is required"
level=[W] time=2021-06-30T06:25:48Z program=any_driver_HEAD source=spec_runner message="Missing hash key: "title" (KeyError)
  from /usr/share/crystal/src/hash.cr:1027:11 in '[]'
  from /usr/share/crystal/src/json/any.cr:102:7 in '[]'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:148:62 in '__build_request__'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:95:17 in '__exec_request__'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:12:5 in '???'
  from repositories/custom_drivers/drivers/events/any_driver.cr:56:22 in 'check_booking'
  from repositories/custom_drivers/drivers/events/any_driver.cr:8:1 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'execute'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:131:5 in 'execute'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:153:18 in 'process'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:141:9 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'run'
  from /usr/share/crystal/src/fiber.cr:92:34 in '->'
  from ???
"
level=[E] time=2021-06-30T06:25:48Z program=any_driver_HEAD source=spec_runner message="executing {"__exec__":"check_booking","check_booking":["{\"action\":\"created\",\"event_id\":\"AAkALgAAAAAAHYQDEapmEc2byACqAC-EWg0A-QGBc8LokEuysfeVGVmyMAAABv0b0gAA\",\"system_id\":\"123qwe\",\"host\":\"Justin.COOK@suncorptest.com.au\",\"resource\":\"system@suncorptest.com.au\",\"ext_data\":{}}"]} on Desk::DeskBookingNotification (spec_runner)"
Missing hash key: "title" (KeyError)
  from /usr/share/crystal/src/hash.cr:1027:11 in '[]'
  from /usr/share/crystal/src/json/any.cr:102:7 in '[]'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:148:62 in '__build_request__'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:95:17 in '__exec_request__'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:12:5 in '???'
  from repositories/custom_drivers/drivers/events/any_driver.cr:56:22 in 'check_booking'
  from repositories/custom_drivers/drivers/events/any_driver.cr:8:1 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'execute'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:131:5 in 'execute'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:153:18 in 'process'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:141:9 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'run'
  from /usr/share/crystal/src/fiber.cr:92:34 in '->'
  from ???

level=ERROR : unhandled exception in spec
Missing hash key: "title" (KeyError) (PlaceOS::Driver::RemoteException)
  from /usr/share/crystal/src/hash.cr:1027:11 in '[]'
  from /usr/share/crystal/src/json/any.cr:102:7 in '[]'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:148:62 in '__build_request__'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:95:17 in '__exec_request__'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/proxy/driver.cr:12:5 in '???'
  from repositories/custom_drivers/drivers/events/any_driver.cr:56:22 in 'check_booking'
  from repositories/custom_drivers/drivers/events/any_driver.cr:8:1 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'execute'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:131:5 in 'execute'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:153:18 in 'process'
  from repositories/custom_drivers/lib/placeos-driver/src/placeos-driver/driver_manager.cr:141:9 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'run'
  from /usr/share/crystal/src/fiber.cr:92:34 in '->'
```

**Versions (please complete the following information):**

- Crystal version:
Crystal 1.0.0 (2021-03-22)

LLVM: 9.0.1
Default target: x86_64-apple-macosx
- drivers:
    git: https://github.com/placeos/drivers.git
    version: 2.2.0+git.commit.d287e3cd30e1d6f2552a995b745e70595004fca1

**Comment by caspiano** [comment]
Thanks for the detailed report!

**Comment by stakach** [comment]
this is resolved in placeos-driver v5.4.5
run `shards update --ignore-crystal-version` in your drivers repository to resolve :)
